roles/openshift_health_checker/library/rpm_version.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+# pylint: disable=missing-docstring
+"""
+Ansible module for rpm-based systems determining existing package version information in a host.
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+IMPORT_EXCEPTION = None
+try:
+    import rpm  # pylint: disable=import-error
+except ImportError as err:
+    IMPORT_EXCEPTION = err  # in tox test env, yum import fails
+
+
+LIST_PKG_NAME = 0
+LIST_PKG_VERSION = 1
+
+
+class RpmVersionException(Exception):
+    """Base exception class for package version przoblems"""
+    def __init__(self, message, problem_pkgs=None):
+        Exception.__init__(self, message)
+        self.problem_pkgs = problem_pkgs
+
+
+def main():
+    """Entrypoint for this Ansible module"""
+    module = AnsibleModule(
+        argument_spec=dict(
+            package_list=dict(type="list", required=True),
+        ),
+        supports_check_mode=True
+    )
+
+    if IMPORT_EXCEPTION:
+        module.fail_json(msg="rpm_version module could not import rpm: %s" % IMPORT_EXCEPTION)
+
+    # determine the packages we will look for
+    pkg_list = module.params['package_list']
+    if not pkg_list:
+        module.fail_json(msg="package_list must not be empty")
+
+    # get list of packages available and complain if any
+    # of them are missing or if any errors occur
+    try:
+        pkg_versions = _retrieve_expected_package_versions(_to_dict(pkg_list))
+        _check_pkg_versions(pkg_versions, _to_dict(pkg_list))
+    except RpmVersionException as excinfo:
+        module.fail_json(msg=str(excinfo))
+    module.exit_json(changed=False)
+
+
+def _to_dict(pkg_list):
+    return {pkg[LIST_PKG_NAME]: pkg for pkg in pkg_list}
+
+
+# pylint: disable=invalid-name
+def _retrieve_expected_package_versions(expected_pkgs_dict):
+    """Search for installed packages matching given pkg names
+    and versions. Returns a dictionary: {pkg_name: [versions]}"""
+
+    ts = rpm.TransactionSet()  # pylint: disable=invalid-name
```

Comment:
https://github.com/openshift/openshift-ansible/blob/master/docs/best_practices_guide.adoc#All-PyLint-rule-disables-MUST-be-documented-in-the-code